File: CPP/Dandelion.cpp
#include "Dandelion.h"

namespace World {
	namespace Plants {

		void Dandelion::spawn_new(const int x, const int y) {
			Organism::spawn_new(x, y, new Dandelion(x, y));
		}

		// 3 próby rozsiania
		void Dandelion::action() {
			for (size_t i = 0; i < 3; ++i) Plant::action();
		}

	}
}

File: java/main/src/pl/vot/tomekby/World/Plants/Dandelion.java
package pl.vot.tomekby.World.Plants;

import pl.vot.tomekby.World.Organism;
import pl.vot.tomekby.World.World;

/**
 * Created by Tomasz on 2015-05-07.
 */
final public class Dandelion extends Plant {
    public Dandelion() { super(); init(); }
    public Dandelion(int x, int y) { super(x, y); init(); }
    public Dandelion(int x, int y, World world) { super(x, y, world); init(); }

    public String get_name() { return "Mlecz"; }
    public void action(Organism with) {
        for(int i = 0; i < 3; ++i) super.action();
    }
    protected void spawn_new(int x, int y) {
        super.spawn_new(new Dandelion(x, y));
    }
    protected void init() {
        strength = 0;
        SYMBOL = 'm';
    }
}
